WalletWasabi/BitcoinCore/Rpc/CachedRpcClient.cs

```diff
@@ -27,8 +27,9 @@ private CancellationTokenSource TipChangeCancellationTokenSource
 		{
 			lock (CancellationTokenSourceLock)
 			{
-				if (_tipChangeCancellationTokenSource is null || _tipChangeCancellationTokenSource.IsCancellationRequested)
+				if (_tipChangeCancellationTokenSource.IsCancellationRequested)
 				{
+					_tipChangeCancellationTokenSource.Dispose();
```

Comment:
I think this is dangerous. Somehow we need to ensure that the tasks that are relying on this cancellationToken are finished, otherwise, they will get `objectdisposedexception`.

Reply:
If we store the reference to the Token itself as we do here:

https://github.com/zkSNACKs/WalletWasabi/pull/7542/files#diff-c00140bebab2e52aea8defc5d0bcdf7c0c0e832aefe5200f0991c5e35ad57e46L175

the CancellationTokenSource can be disposed and anyone who is using the Token won't have problems. Only calls to the tokensource itself will generate ObjectDisposedExceptions. 
Source https://www.rocksolidknowledge.com/articles/working-with-cancellationtoken-and-dispose.